cns/restserver/api.go

```diff
@@ -1461,7 +1494,11 @@ func (service *HTTPRestService) nmAgentSupportedApisHandler(w http.ResponseWrite
 		if err != nil {
 			returnCode = types.NmAgentSupportedApisError
 			returnMessage = fmt.Sprintf("[Azure-CNS] %s", retErr.Error())
+		} else {
+			// caching supportedApis value
+			service.updateSupportedApisCache(apis)
```

Comment:
I think we should cache the support APIs. and we will not miss out the underneath version update on nmagent because when we find out the api list cache does not contain the target api, we will make call to get the updated api list from nmagent and take different action based on the response